src/coreclr/jit/emitriscv64.h

```diff
@@ -81,6 +81,12 @@ bool IsRedundantLdStr(
 /************************************************************************/
 
 public:
+// Returns true if 'value' is a legal signed immediate 13 bit encoding.
+static bool isValidSimm13(ssize_t value)
+{
+    return -(((int)1) << 12) <= value && value < (((int)1) << 12);
```

Comment:
13 bit and 21 bit immediates should have `0` in least bit. Should we check it?

Reply:
In most cases, there are checks. I added one. They check least two bits are 0 because there is no compact instructions now.